pkg/backend/local/progress.go

```diff
@@ -407,18 +408,18 @@ func (display *ProgressDisplay) getOrCreateTreeNode(
 		}
 
 		parentRow, hasParentRow := display.eventUrnToResourceRow[parentURN]
+
 		if !hasParentRow {
-			// if we couldn't find a parent for this yet, parent this to the stack if we
-			// have an entry for that.  Otherwise, we'll just make this a top level item.
+			// If we haven't heard about this node's parent, then  just parent it to the stack.
+			// Note: getting the parent row for the stack-urn will always succeed as we ensure that
+			// such a row is always there in ensureHeaderAndStackRows
 			parentURN = display.stackUrn
-			parentRow, hasParentRow = display.eventUrnToResourceRow[parentURN]
+			parentRow = display.eventUrnToResourceRow[parentURN]
 		}
 
-		if hasParentRow {
-			parentNode := display.getOrCreateTreeNode(result, parentURN, parentRow, urnToTreeNode)
-			parentNode.childNodes = append(parentNode.childNodes, node)
-			return node
-		}
+		parentNode := display.getOrCreateTreeNode(result, parentURN, parentRow, urnToTreeNode)
+		parentNode.childNodes = append(parentNode.childNodes, node)
+		return node
```

Comment:
i updated this code because hte previous logic kept breaking my brain.  Effectively i wanted to say: for any resource (except the root-stack resource), you should alwyas be able to parent it to another resource.  The easiest way to do this was to ensure we'd always have a root tree node to parent to.  And the easiest way to do that was to ensure we had a root resource-row we'd generate the root node out of.